core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go

```diff
@@ -0,0 +1,496 @@
+}
+
+// TODO Make this return an interpretation error????
+func TranspileDockerComposePackageToStarlark(packageAbsDirpath string, composeRelativeFilepath string) (string, error) {
+	composeAbsFilepath := path.Join(packageAbsDirpath, composeRelativeFilepath)
+
+	// Useful for logging, to not leak internals of APIC
+	composeFilename := path.Base(composeRelativeFilepath)
+
+	composeBytes, err := os.ReadFile(composeAbsFilepath)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred reading Compose file '%v'", composeFilename)
+	}
+
+	// Use the envvars file next to the Compose if it exists
+	envVarsFilepath := path.Join(packageAbsDirpath, envVarsFilename)
+	var envVars map[string]string
+	envVarsInFile, err := godotenv.Read(envVarsFilepath)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return "", stacktrace.Propagate(err, "Failed to transpile Docker Compose package to Starlark; a %v file was detected in the package, but an error occurred reading", envVarsFilename)
+		}
+		envVarsInFile = map[string]string{}
+	}
+	envVars = envVarsInFile
+
+	script, err := convertComposeToStarlark(composeBytes, envVars)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred transpiling Compose file '%v' to Starlark", composeFilename)
+	}
+	return script, nil
+}
+
+// ====================================================================================================
+//                                   Private Helper Functions
+// ====================================================================================================
+
+func convertComposeToStarlark(composeBytes []byte, envVars map[string]string) (string, error) {
+	composeParseConfig := types.ConfigDetails{ //nolint:exhaustruct
+		// Note that we might be able to use the WorkingDir property instead, to parse the entire directory
+		ConfigFiles: []types.ConfigFile{{
+			Content: composeBytes,
+		}},
+		Environment: envVars,
+	}
+
+	setOptionsFunc := func(options *loader.Options) {
+		options.SetProjectName(composeProjectName, shouldOverrideComposeYamlKeyProjectName)
+
+		// We don't want to resolve paths; these should get resolved by our package content provider instead
+		options.ResolvePaths = false
+
+		// We do want to convert Windows paths into Linux paths, since the APIC runs on Linux
+		options.ConvertWindowsPaths = true
+	}
+
+	compose, err := loader.Load(composeParseConfig, setOptionsFunc)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred parsing the Compose file in preparation for Starlark transpilation")
+	}
+
+	perServiceDependencies := map[string]map[string]bool{}
+
+	// Mapping of services -> lines of Starlark that they need to run
+	perServiceLines := map[string][]string{}
+	numTotalServiceLines := 0
+
+	// List of sorted service names that we'll use to make sure we process depends_on correctly
+	sortedServiceNames := []string{}
+
+	for _, serviceConfig := range compose.Services {
```

Comment:
rename `serviceConfig` to `composeServiceConfig` to distinguish from a starlark `ServiceConfig`